packages/firebase_ml/lib/src/model_management/firebase_model_manager.dart

```diff
@@ -18,32 +19,35 @@ import 'firebase_model_download_conditions.dart';
 /// https://firebase.google.com/docs/reference/android/com/google/
 /// firebase/ml/common/modeldownload/FirebaseModelManager
 class FirebaseModelManager {
-  static const MethodChannel _channel =
+  FirebaseModelManager._();
+
+  @visibleForTesting
+
+  /// Means for communication with native platform code
+  static const MethodChannel channel =
       MethodChannel('plugins.flutter.io/firebase_ml');
 
   /// Singleton of [FirebaseModelManager].
-  static final FirebaseModelManager instance = FirebaseModelManager();
+  static final FirebaseModelManager instance = FirebaseModelManager._();
 
   /// Initiates the download of remoteModel if the download hasn't begun.
   Future<void> download(FirebaseRemoteModel model,
       FirebaseModelDownloadConditions conditions) async {
-    // Example call
-    await _channel.invokeMethod(
-        "FirebaseModelManager#download", {'model': model.toString()});
+    Map modelMap = await channel.invokeMethod("FirebaseModelManager#download",
+        {'model': model.modelName, 'conditions': conditions.toMap()});
+    model.modelHash = modelMap['modelHash'];
   }
 
-  /// Deletes the given [FirebaseRemoteModel] from disk.
-  Future<void> deleteDownloadedModel(FirebaseRemoteModel model) async {}
-
   /// Returns the [File] containing the latest model for the remote model name.
-  Future<File> getLatestModelFile(
-      FirebaseRemoteModel firebaseRemoteModel) async {
-    return File("");
+  Future<File> getLatestModelFile(FirebaseRemoteModel model) async {
+    String modelPath = await channel.invokeMethod(
+        "FirebaseModelManager#getLatestModelFile", {'model': model.modelName});
+    return File(modelPath);
   }
 
   /// Returns whether the given [FirebaseRemoteModel] is currently downloaded.
-  Future<bool> isModelDownloaded(
-      FirebaseRemoteModel firebaseRemoteModel) async {
-    return true;
+  Future<bool> isModelDownloaded(FirebaseRemoteModel model) async {
+    return await channel.invokeMethod(
+        "FirebaseModelManager#isModelDownloaded", {'model': model.modelName});
```

Comment:
Does this need to check against modelHash to make sure that version of the model is downloaded?

Reply:
It doesn't explicitly say in the documentation if the model is checked by the model hash. To me it seems like it's checked by the name only. Would be useful to check in the SDK. I can also pass hash just in case.
"Downloads the given model from the server to a local directory on the device. Use isModelDownloaded(_:) to check the download status for the model. If this method is invoked and the model has already been downloaded, a request is made to check if a newer version of the model is available for download. If available, the new version of the model is downloaded."
-- from IOS docs

Reply:
from the documentation it doesn't look like modelHash is used, maybe worth double checking. I think the API would be nicer if we can just use `modelName` alone in the input params

Reply:
I would keep it as it is to be consistent with native APIs. I would really like to know why Java and Obj-C APIs require full model and not only the name.
If we end up changing it, it would make sense to change getLatestModelFile as well.
I would appreciate @bparrishMines and @nauthadr input on this.

Reply:
I would be in favour of being consistent with the native APIs. Also, if we are undergoing a Firebase counsel API review may make sense to leave as is in this PR for now and then change accordingly based on feedback from the API review.

Reply:
+1 for waiting for the API review